src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -6661,6 +6661,12 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="WRN_CallerArgumentExpressionAttributeHasInvalidParameterName_Title" xml:space="preserve">
     <value>The CallerArgumentExpressionAttribute is applied with an invalid parameter name.</value>
   </data>
+  <data name="WRN_CallerArgumentExpressionAttributeSelfReferential" xml:space="preserve">
+    <value>The CallerArgumentExpressionAttribute applied to parameter '{0}' will have no effect because it's self-referential.</value>
+  </data>
+  <data name="WRN_CallerArgumentExpressionAttributeSelfReferential_Title" xml:space="preserve">
+    <value>The CallerArgumentExpressionAttribute is applied with a self-referential parameter.</value>
```

Comment:
>The CallerArgumentExpressionAttribute is applied with a self-referential parameter. [](http://example.com/codeflow?start=11&length=83)

We usually use the original message with removed placeholders. i think this will work well in this case as well: "The CallerArgumentExpressionAttribute applied to parameter will have no effect because it's self-referential."